plugins/rbenv/rbenv.plugin.zsh

```diff
@@ -1,60 +1,69 @@
+
+FOUND_RBENV=$+commands[rbenv]
+
+if [[ $FOUND_RBENV -ne 1 ]]; then
+    rbenvdirs=("$HOME/.rbenv" "/usr/local/rbenv" "/opt/rbenv" "/usr/local/opt/rbenv")
+    for dir in $rbenvdirs; do
+        if [[ -d $dir/shims ]]; then
+            export PATH="$dir/shims:$dir/bin:$PATH"
+            FOUND_RBENV=1
+            break
+        fi
+    done
+fi
+
+if [[ $FOUND_RBENV -ne 1 ]]; then
+    if (( $+commands[brew] )) && dir=$(brew --prefix rbenv 2>/dev/null); then
+        if [[ -d $dir/shims ]]; then
+            export PATH="$dir/shims:$dir/bin:$PATH"
+            export RBENV_ROOT=$dir
+            FOUND_RBENV=1
+        fi
     fi
 fi
 
-for rbenvdir in "${rbenvdirs[@]}" ; do
-  if [ -d $rbenvdir/bin -a $FOUND_RBENV -eq 0 ] ; then
-    FOUND_RBENV=1
+if [[ $FOUND_RBENV -eq 1 ]]; then
     if [[ $RBENV_ROOT = '' ]]; then
-      RBENV_ROOT=$rbenvdir
+      export RBENV_ROOT="$HOME/.rbenv"
     fi
-    export RBENV_ROOT
-    export PATH=${rbenvdir}/bin:$PATH
     eval "$(rbenv init --no-rehash - zsh)"
 
     alias rubies="rbenv versions"
     alias gemsets="rbenv gemset list"
 
     function current_ruby() {
-      echo "$(rbenv version-name)"
+        echo "$(rbenv version-name)"
     }
 
     function current_gemset() {
-      echo "$(rbenv gemset active 2&>/dev/null | sed -e ":a" -e '$ s/\n/+/gp;N;b a' | head -n1)"
+        echo "$(rbenv gemset active 2&>/dev/null | sed -e ":a" -e '$ s/\n/+/gp;N;b a' | head -n1)"
     }
 
     function gems {
-      local rbenv_path=$(rbenv prefix)
-      gem list $@ | sed -E \
-        -e "s/\([0-9a-z, \.]+( .+)?\)/$fg[blue]&$reset_color/g" \
-        -e "s|$(echo $rbenv_path)|$fg[magenta]\$rbenv_path$reset_color|g" \
-        -e "s/$current_ruby@global/$fg[yellow]&$reset_color/g" \
-        -e "s/$current_ruby$current_gemset$/$fg[green]&$reset_color/g"
+        local rbenv_path=$(rbenv prefix)
+        gem list $@ | sed -E \
+          -e "s/\([0-9a-z, \.]+( .+)?\)/$fg[blue]&$reset_color/g" \
+          -e "s|$(echo $rbenv_path)|$fg[magenta]\$rbenv_path$reset_color|g" \
+          -e "s/$current_ruby@global/$fg[yellow]&$reset_color/g" \
+          -e "s/$current_ruby$current_gemset$/$fg[green]&$reset_color/g"
     }
 
     function rbenv_prompt_info() {
-      if [[ -n $(current_gemset) ]] ; then
-        echo "$(current_ruby)@$(current_gemset)"
-      else
-        echo "$(current_ruby)"
-      fi
+        if [[ -n $(current_gemset) ]] ; then
+            echo "$(current_ruby)@$(current_gemset)"
+        else
+            echo "$(current_ruby)"
+        fi
     }
-  fi
-done
-unset rbenvdir
-
-if [ $FOUND_RBENV -eq 0 ] ; then
-  alias rubies='ruby -v'
-  function gemsets() { echo 'not supported' }
-  function rbenv_prompt_info() { echo "system: $(ruby -v | cut -f-2 -d ' ')" }
+else
+    alias rubies='ruby -v'
+    alias gemsets='echo'
+    function current_ruby() { echo 'not supported' }
+    function current_gemset() { echo 'not supported' }
+    function gemsets() { echo 'not supported' }
```

Comment:
Here and above: how about double quotes for consistency?

Reply:
Done.